File: frontend/src/lib/types/resource.ts
export interface Resource {
  id: string; // UUID
  organization_id: string; // UUID
  cloud_account_id: string; // UUID
  provider_resource_id: string;
  name: string | null;
  resource_type: string;
  region: string;
  state: string;
  tags: Record<string, string> | null;
  instance_type: string | null;
  provider_type: string | null;
  last_seen_at: string; // ISO datetime
  created_at: string; // ISO datetime
  updated_at: string; // ISO datetime
}

export interface ResourceFilter {
  provider_type?: string;
  state?: string;
  cloud_account_id?: string;
  region?: string;
  page?: number;
  page_size?: number;
}

export interface PaginatedResources {
  items: Resource[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

File: app/schemas/resource.py
from pydantic import BaseModel
from typing import Optional, Dict, List
from datetime import datetime
import uuid


class ResourceBase(BaseModel):
    provider_resource_id: str
    name: Optional[str] = None
    region: str
    state: str
    tags: Optional[Dict[str, str]] = None
    instance_type: Optional[str] = None


class Resource(ResourceBase):
    id: uuid.UUID
    organization_id: uuid.UUID
    cloud_account_id: uuid.UUID
    resource_type: str
    provider_type: Optional[str] = None
    last_seen_at: datetime
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class PaginatedResources(BaseModel):
    items: List[Resource]
    total: int
    page: int
    page_size: int
    total_pages: int


class ResourceFilter(BaseModel):
    """Filters for listing resources."""
    provider_type: Optional[str] = None
    state: Optional[str] = None
    cloud_account_id: Optional[uuid.UUID] = None
    region: Optional[str] = None
